js/main.js

```diff
@@ -1,45 +1,491 @@
 function createBoard(str) {
-  boardDivs.forEach((indArr, outerIdx) => {
+  PLAYER_BOARD.forEach((indArr, outerIdx) => {
     indArr.forEach((arrEl, innerIdx) => {
       const div = document.createElement("DIV");
-      div.className = "board";
-      if (innerIdx === 0) {
-        div.innerHTML = 'c' + (outerIdx+1);
-      } else
-        div.innerHTML = `c${outerIdx + 1}r${String.fromCharCode(
-          "a".charCodeAt(0) + innerIdx - 1
-        )}`;
+      div.innerHTML = "";
+      //Used to start at row A and increment to the next character in the alphabet
+      div.className = `board ${String.fromCharCode(
+        "a".charCodeAt(0) + innerIdx
+      )}${outerIdx + 1}`;
+      //Placing an active game space into an array to be accessed
+      if (parseInt(div.innerHTML) !== outerIdx + 1) {
+        str === "player" ? PLAYER_DIVS.push(div) : CPU_DIVS.push(div);
+      }
+      // Checking if the createBoard function was called for the player or the board to append the divs to the correct grid
       str === "player"
-        ? playerBoardSection.appendChild(div)
-        : cpuBoardSection.appendChild(div);
+        ? PLAYER_BOARD_SECTION.prepend(div)
+        : CPU_BOARD_SECTION.appendChild(div);
     });
   });
 }
+
+//Handling the player clicks for placing ships
+function playerMouseClick(evt) {
+  if (evt.target.tagName === "INPUT" || evt.target.tagName === "LABEL" || shipCounter > 6) return;
+  let clickedDiv = PLAYER_DIVS.indexOf(evt.target);
+  let rows = parseInt(clickedDiv.toString().charAt(0));
+  let columns = parseInt(clickedDiv.toString().charAt(1));
+  if (isNaN(columns)) {
+    columns = rows;
+    rows = 0;
+  }
+  let shipCanBePlaced;
+  shipCanBePlaced = canBePlaced(
+    PLAYER_BOARD,
+    rows,
+    columns,
+    verticalCheckbox === true ? "vertical" : "horizontal",
+    Object.values(playerShips)[shipCounter]
+  );
+  if (shipCanBePlaced === true) {
+    placeShips(
+      PLAYER_BOARD,
+      rows,
+      columns,
+      verticalCheckbox === true ? "vertical" : "horizontal",
+      Object.values(playerShips)[shipCounter]
+    );
+    shipCounter++;
+  }
+  activeTurn();
+}
+
+//Rendering the ships on the game board
+function renderShips(board) {
+  let shipCanBePlaced;
+  let rows;
+  let columns;
+  let idx = Math.round(Math.random());
+  const vertOrHoriz = ["vertical", "horizontal"];
+
+  Object.values(cpuShips).forEach((shipObj) => {
+    do {
+      rows = getRandomNum(0, HEIGHT);
+      columns = getRandomNum(0, LENGTH);
+      shipCanBePlaced = canBePlaced(
+        board,
+        rows,
+        columns,
+        vertOrHoriz[idx],
+        shipObj
+      );
+    } while (!shipCanBePlaced);
+    placeShips(board, rows, columns, vertOrHoriz[idx], shipObj);
+    idx = Math.round(Math.random());
+  });
+}
+
+//Modifying the header based on the current games state
+function activeTurn() {
+  if (shipCounter <= 6) {
+    activePlayerEl.innerHTML = `Place your ${
+      Object.values(playerShips)[shipCounter].name
```

Comment:
`Object.values` will loop through an object and create an array of values each time you call it. In this scenario you're calling it twice to create this sting. Instead you can assign the array to a temp variable so that you only have to call `Object.values` once. Something like:

`
const ship = Object.values(playerShips)[shipCounter];
`

And then just to `ship.name` and `ship.length`